musl doesn't have res_nquery - fall back to res_query
--- a/netwerk/dns/PlatformDNSUnix.cpp
+++ b/netwerk/dns/PlatformDNSUnix.cpp
@@ -32,11 +32,11 @@
   rv = packet.FillBuffer(
       [&](unsigned char response[DNSPacket::MAX_SIZE]) -> int {
         int len = 0;
-#if defined(XP_LINUX)
+#if defined(__GLIBC__)
         len = res_nquery(&_res, host.get(), ns_c_in,
                          nsIDNSService::RESOLVE_TYPE_HTTPSSVC, response,
                          DNSPacket::MAX_SIZE);
-#elif defined(XP_MACOSX)
+#elif defined(XP_LINUX) || defined(XP_MACOSX)
         len =
             res_query(host.get(), ns_c_in, nsIDNSService::RESOLVE_TYPE_HTTPSSVC,
                       response, DNSPacket::MAX_SIZE);
